src/components/location-search/location-search.tsx

```diff
@@ -0,0 +1,118 @@
+import debounce from 'lodash.debounce';
+import { ComboBox, ESystemIconNames } from '@mdb/flora';
+import { ChangeEvent, useCallback, useMemo, useState } from 'react';
+
+import { getURLPath } from '../../utils/format-url-path';
+
+type Option = string | { icon: string; label: string };
+
+// TODO: needs to be updated with icon names that will be added by Web Team - https://jira.mongodb.org/browse/WEBSITE-13740
+const DEFAULT_OPTIONS = [
+    { icon: ESystemIconNames.ARROW_UP, label: 'Current Location' },
+    { icon: ESystemIconNames.ARROW_DOWN, label: 'Virtual' },
+];
+
+interface LocationSearchProps {
+    onSelectCallback?: (selection: string) => void;
+}
+
+export default function LocationSearch({
+    onSelectCallback = () => null,
+}: LocationSearchProps) {
+    const [value, setValue] = useState('');
+    const [isLoading, setIsLoading] = useState(false);
+    const [options, setOptions] = useState<Option[]>(DEFAULT_OPTIONS);
+
+    async function geolocationSuccessCb({
+        coords: { latitude, longitude },
+    }: {
+        coords: GeolocationCoordinates;
+    }) {
+        const req = await fetch(
+            `${getURLPath('api/geocode')}?latlng=${latitude},${longitude}`
+        );
+        const { error, results } = await req.json();
+
+        if (error) {
+            // There is no visual error handling planned to display to user, so just log it in the console
+            console.warn(error);
+            setValue('');
+        } else {
+            setValue(results[0]?.formatted_address);
+        }
+    }
+
+    function geolocationErrorCb() {
+        console.warn(
+            'Cannot get current location, as location sharing is disabled'
+        );
+        setValue('');
+    }
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    const debouncedSearch = useCallback(
+        debounce(async value => {
+            const req = await fetch(
+                `${getURLPath('api/location')}?search=${value}`
+            );
+            const { results, error } = await req.json();
+
+            if (error) {
+                // There is no visual error handling planned to display to user, so just log it in the console
+                console.warn(error);
+                setOptions([]);
+            } else {
+                setOptions(
+                    results.map(
+                        ({ description }: { description: string }) =>
+                            description
+                    )
+                );
+            }
+
+            setIsLoading(false);
+        }, 500),
```

Comment:
this debounce ms value differs from the one we use in the search field [here](https://github.com/mongodb/devcenter/blob/68e1df61db432b891e0b4f97b21a9f043492ab63/src/hooks/search/index.ts#L99). might be worth moving this to a constant and sharing it between the two.